modules/currency_denomination/commerce_pos_currency_denomination.install

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/**
+ * @file
+ * Schema and install hooks for commerce_pos_currency_denomination.
+ */
+
+/**
+ * Implements hook_schema().
+ */
+function commerce_pos_currency_denomination_schema() {
+  $schema['commerce_pos_currency_denomination_type'] = array(
+    'description' => 'The base table for a currency denomination type.',
+    'fields' => array(
+      'currency_id' => array(
+        'description' => 'The primary identifier for a currency denomination type.',
+        'type' => 'serial',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+      ),
+      'currency_code' => array(
+        'description' => 'The human-readable name of the currency denomination type.',
+        'type' => 'varchar',
+        'length' => 255,
+        'not null' => TRUE,
+        'default' => '',
+      ),
+    ),
+    'primary key' => array('currency_id'),
+  );
+
+  $schema['commerce_pos_currency_denomination'] = array(
+    'description' => 'The base table for adding a currency denomination.',
+    'fields' => array(
+      'denomination_id' => array(
+        'description' => 'The primary identifier for a currency denomination.',
+        'type' => 'serial',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+      ),
+      'currency_id' => array(
+        'description' => 'The primary identifier from the currency denomination type table.',
+        'type' => 'int',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+        'default' => 0,
+      ),
+      'denomination_name' => array(
+        'description' => 'The machine name of the denomination eg. penny.',
+        'type' => 'varchar',
+        'length' => 255,
+        'not null' => TRUE,
+        'default' => '',
+      ),
+      'denomination_title' => array(
+        'description' => 'The human-readable name of the denomination eg. Penny.',
+        'type' => 'varchar',
+        'length' => 255,
+        'not null' => TRUE,
+        'default' => '',
+      ),
+      'denomination_amount' => array(
+        'description' => 'The amount of the currency denomination eg. 1.',
+        'type' => 'int',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+        'default' => 0,
+      ),
+    ),
+    'indexes' => array(
+      'currency_id' => array('currency_id'),

```

Comment:
I suggest adding 'denomination_name' as an index as well, since it is used as a condition in a query below. In fact, perhaps making a unique key consisting of currency_id and denomination_name would be better... At the very least though we can make denomination_name an index.